data/src/main/java/io/gnosis/data/repositories/SafeRepository.kt

```diff
@@ -49,9 +49,9 @@ class SafeRepository(
             isSupported(request.checkedResult("Valid safe check failed").asEthereumAddress())
         }
 
-    suspend fun setActiveSafe(safe: Safe) {
+    suspend fun setActiveSafe(safe: Safe?) {
         preferencesManager.prefs.edit {
-            putString(ACTIVE_SAFE, safe.address.asEthereumAddressString())
+            putString(ACTIVE_SAFE, safe?.address?.asEthereumAddressString())
```

Comment:
I think this is the root of the change of the patter. I think it would be mucho better to have an additional method `removeSafe()` that sets the active safe to null, than making this nullable. We lose so much null safety just by this design decision.

Reply:
What about tests for this in the `SafeRepositoryTest.kt`? I would sincerely ask you to not make this nullable. The null semantics are not always obvious, it's much more preferable to set and explicit `remove` method for this.

Reply:
activeSafe == null is a valid case. Having additional removeSafe just adds complexity.